docs/release-notes/rl-0.8.md

```diff
@@ -28,6 +28,61 @@
   align with the "hunks" themed mapping and avoid conflict with the new [neogit]
   group.
 
+- Language modules are reworked to allow more flexible and close-to-neovim-API
+  configs. LSP configuration options are removed from `vim.languages` and should
+  be done via `vim.lsp.servers`.
+
+## Language module LSP rework
+
+in nvf v0.8, language modules are overhauled, with a few goals in mind:
+
+1. get rid of our rigid "hard-coded" lua scripts, in favor of the more flexible
+   `setupOpts` styled options
+2. have our new API stick closer to lua APIs, in this case `vim.lsp.config`
+3. allow enabling multile LSPs per-language
+
+This is, however, a big change, so breaking changes are inevitable. The most
+important change is the removal of `vim.languages.<lang>.lsp.package` - you now
+need to modify `vim.lsp.servers.<server>.cmd`, copying over any additional
+arguments - these can be found in the source code files of the respective
+language.
+
+> [!NOTE] The `<server>` mentioned here is the name of the language server, as
+> shown in `vim.languages.<lang>.lsp.servers`. This is NOT the name of the
+> language.
```

Comment:
The current documentation framework won't be able to parse GFM callouts. Maybe don't use `NOTE` here yet, but leave aa comment so that we remember to update it once I migrate docs to ndg.